### PR TITLE
Add TestOverride.SweepTimeout

### PR DESCRIPTION
**Pull request by markfields**
## Background

As-is, the Sweep Timeout will always be at least 6 days due to hardcoded Snapshot Cache limit (5 days) and Buffer (1 day).  This means it's impossible to test the Sweep Timer and the code that runs after something is SweepReady!

There have been multiple attempts on my part to address this:  #11084, #12044, #12331

Much of the difficulty has arisen from the challenge of aligning driver policy with GC config.  I'm finally punting on that and tracking it as a separate item ([AB#2238](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/2238)).

## This change

It's pretty simple:

* Persist SweepTimeoutMs in the summary
* For new containers, that can be overridden via the `"Fluid.GarbageCollection.TestOverride.SweepTimeoutMs"` config setting
* Default behavior for new containers (and for backfilling existing containers with no value persisted) is SweepTimeout + 6 days (same as it has been)

### CODE REVIEW

File: packages/runtime/container-runtime/src/garbageCollection.ts [hunk 1]
```diff
@@ -498,6 +496,21 @@ export class GarbageCollector implements IGarbageCollector {
...
+        /**
+         * Sweep timeout is the time after which unreferenced content can be swept.
+         * Sweep timeout = session expiry timeout + snapshot cache expiry timeout + one day buffer.
+         *
+         * The snapshot cache expiry timeout cannot be known precisely but the upper bound is 5 days.
+         * The buffer is added to account for any clock skew or other edge cases.
+         * We use server timestamps throughout so the skew should be minimal but make it 1 day to be safe.
+         */
+        function computeSweepTimeout(sessionExpiryTimeoutMs: number | undefined) {
+            const maxSnapshoCacheExpiryMs = 5 * oneDayMs;
```
Comment (tyler-cai-microsoft) [inline packages/runtime/container-runtime/src/garbageCollection.ts:508]: Nit: `maxSnapsho(t)CacheExpiryMs` missing the `t` between the `o` and the `C`.

File: packages/runtime/container-runtime/src/garbageCollection.ts [hunk 2]
```diff
@@ -252,6 +249,7 @@ export class UnreferencedStateTracker {
         this.sweepTimer = new TimerWithNoDefaultTimeout(
             () => {
                 this._state = UnreferencedState.SweepReady;
+                //* Follow up on race condition here when timeouts are sub-second
```
Comment (markfields) [inline packages/runtime/container-runtime/src/garbageCollection.ts:252]: Todo - open a bug and delete this comment

File: packages/runtime/container-runtime/src/garbageCollection.ts [hunk 3]
```diff
@@ -584,16 +593,16 @@ export class GarbageCollector implements IGarbageCollector {
...
          * feature flag.
          */
-        this.shouldRunSweep = false; // disable while TEMPORARY measure hardcoding snapshotCacheExpiryMs is here
-            // this.shouldRunGC
-            // && this.sweepTimeoutMs !== undefined
-            // && (this.mc.config.getBoolean(runSweepKey) ?? this.sweepEnabled);
+        this.shouldRunSweep =
+            this.shouldRunGC
+            && this.sweepTimeoutMs !== undefined
+            && (this.mc.config.getBoolean(runSweepKey) ?? this.sweepEnabled);
```
Comment (tyler-cai-microsoft) [inline packages/runtime/container-runtime/src/garbageCollection.ts:605]: `sweepEnabled` -> `sweepForceEnabled`?

Comment (markfields) [reply packages/runtime/container-runtime/src/garbageCollection.ts:605]: Can you say more why you're suggeting that?  I'd more say the config setting is the "forceful" one - `this.sweepEnabled` is the value computed on create and written to the file in the initial snapshot.

**Review by tyler-cai-microsoft — APPROVED** [review]
LGTM

File: packages/runtime/container-runtime/src/garbageCollection.ts (hunk 3 above)
Comment (tyler-cai-microsoft) [reply packages/runtime/container-runtime/src/garbageCollection.ts:605]: Specifically `testOverrideSweepTimeoutMs !== undefined` is in the condition of `sweepEnabled`, maybe I'd rather put that in `this.shouldRunSweep`

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 4]
```diff
@@ -995,6 +995,14 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         const pendingRuntimeState = context.pendingLocalState as IPendingRuntimeState | undefined;
         const baseSnapshot: ISnapshotTree | undefined = pendingRuntimeState?.baseSnapshot ?? context.baseSnapshot;
 
+        const maxSnapshotCacheDurationMs = this._storage?.policies?.maximumCacheDurationMs;
+        if (maxSnapshotCacheDurationMs !== undefined && maxSnapshotCacheDurationMs > 5 * 24 * 60 * 60 * 1000) {
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:999]: Consider moving this to GarbageCollector so we have the GC logic in one place.

Comment (markfields) [reply packages/runtime/container-runtime/src/containerRuntime.ts:999]: Hmm maybe. So we'd take in the param still but just for validation?

**Review by agarwal-navin — APPROVED** [review]

File: packages/runtime/container-runtime/src/garbageCollection.ts (hunk 3 above)
Comment (markfields) [reply packages/runtime/container-runtime/src/garbageCollection.ts:605]: It has to go here so the values persisted to the file are coherent.  shouldRunSweep is about this session, not the file's capabilities. I'm inclined to leave it as-is.

**Pull request merged**